corehq/apps/app_manager/views/download.py

```diff
@@ -266,6 +266,7 @@ def create_build_files_if_necessary_handling_conflicts(is_retry=False):
         except ResourceConflict:
             if is_retry:
                 raise
+            request.app = Application.get(request.app.get_id)
```

Comment:
would it be better if this is moved up at the beginning of the method?
```
if retry:
    request.app = Application.get(request.app.get_id)
```

Reply:
functionally that would be the same and it adds an extra conditional. I also think this is easier to reason about since you see exactly why the app is being fetched and you don't have to trace the 'is_retry' var back to see when it would be true.

Reply:
the advantage is see is that one would not need to reload the app each time before calling this method with retry True.
This is fine too.

Reply:
The whole method exists solely for the purpose or retrying after a resource conflict. This is the only place this method is ever called and it's a 'private' method so can't be called anywhere else.